tasmota/tasmota_xsns_sensor/xsns_14_sht3x.ino

```diff
@@ -106,14 +106,14 @@ bool Sht3xRead(uint32_t sensor) {
   if ((Sht3xComputeCrc(&data[0], 2) != data[2]) || (Sht3xComputeCrc(&data[3], 2) != data[5])) {
     return false;
   }
-  float t = ((((data[0] << 8) | data[1]) * 175) / 65535.0f) - 45;
+  int32_t t_100 = ((((data[0] << 8) | data[1]) * 17500) >> 16) - 4500;
```

Comment:
@eku It is now ~1/100th of a degree off compared to the `float`. I can personally live with that. Let's see what @sfromis and @arendst have to say.

I just wanted to mention here that this "issue" with the double precision has not been introduced by me. My motivation for touching this code was to fix the bug https://github.com/arendst/Tasmota/pull/21755 and to reduce the traffic on the bus, not to squeeze the last byte out of the code.

Reply:
LGTM

Reply:
This is a representation of the systematic error introduced for the temperature:
![Untitled 1](https://github.com/user-attachments/assets/ad5b69a8-ba59-457d-bc9d-830df48a0e96)
The median error at data0 = 102, which is approximately 25°C, is round about 0.006 °C. 

This is a representation of the error introduced for the relative humidity:
![Untitled 2](https://github.com/user-attachments/assets/dbacfa1f-e3ef-4271-a08a-777bf768c901)
The median error at data3 = 114, which is approximately 50 %RH, is round about 0.006 %RH.